doc/data/messages/s/simplifiable-if-expression/bad.py

```diff
@@ -0,0 +1,8 @@
+FLYING_THINGS = ["bird", "plane", "superman", "this example"]
+
+def is_flying_animal(an_object):
+    if isinstance(an_object, Animal) and an_object in FLYING_THINGS:
```

Comment:
We need to expect a message here.

Reply:
Yes, it's not raising anything right now, so I did not expect it to work I think it's a problem and that's why this issue is labelled discussion:

> I realized during implementation that simplifiable if expression became worst and that I disagree with https://github.com/PyCQA/pylint/issues/1984 basically. If there are two values but one if the opposite of the other we can pretty much simplify the if expression. It look like the example from the issue come directly from [this stackoverflow question](https://stackoverflow.com/questions/49546992/) but the top answer make a compelling point that this is simplifiable.

Reply:
Ah okay.

I'm not sure whether I think we should raise here. I'll let others give their opinion as well.

Reply:
I think #1684 was correct, because even if the two values are given opposing boolean values, we can't simplify, because they might have had state before.

E.g. this raises `TypeError` because  `b` had state:
```python
def x():
  a = []
  b = []
  if bool('special test'):
    a = True
  else:
    b = False
  return a + b
```

But if we refactor to `return bool('special test')`, then the behavior is different.

Reply:
You're right, thank you for your input 👍 I think this is still simplifiable because clearly the code style is horrible (with type redefinition of the type and the conditional can still be simplified : clearly we don't have this type of construct in the primer 😄) but it's another MR's scope.